config/v1/types_infrastructure.go

```diff
@@ -1769,9 +1769,7 @@ type NutanixFailureDomain struct {
 	// obtained from the Prism Central console or using the prism_central API.
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinItems=1
-	// +kubebuilder:validation:MaxItems=1
-	// +listType=map
-	// +listMapKey=type
+	// +listType=atomic
```

Comment:
Why change to atomic?

Reply:
Because the "map" type requires the listMapKey, and none of the Item fields is suitable to be listMapKey. The "Name" and "UUID" are optional pointer, could have nil value. And the "Type" value is not unique for the list items.

Reply:
Ok, that makes sense, I think we should add CEL though to ensure that the items in the list are unique, something along the lines of
```suggestion
	// +listType=atomic
	// +kubeubilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))",message="each subnet must be unique"
```

Reply:
Changed as suggested.